Add Fusaka FAQ (#16413)
ethereum/ethereum-org-website: public/content/roadmap/fusaka/index.md

### What improvements are included for L2 Scaling? {#what-improvements-are-included-for-l2-scaling}

[PeerDAS](https://eips.ethereum.org/EIPS/eip-7594) is is the main feature of the fork. It implements Data Availability Sampling (DAS). Instead of every full node needing to download and store all blob data (which becomes unsustainable as L2s grow), PeerDAS allows each node to be responsible for only a subset of the blob data while maintaining its availability and verifiability. This enables blob throughput to scale significantly, leading to even smaller fees for Layer 2s.

[Learn more about PeerDAS](/roadmap/fusaka/peerdas)

Alongside PeerDAS, [a small change](https://eips.ethereum.org/EIPS/eip-7918) that improves the blob fee market is being introduced. It ensures that the blob fee market reacts efficiently to congestion and doesn't spiral down to 1 wei, guaranteeing L2s pay a meaningful fee for the compute and space that blobs impose on nodes.

> ### What are Blob Parameter Only forks? {#what-are-blob-parameter-only-forks}

Author: The reason for many of these duplicates is that I originally intended this as a standalone FAQ page. But as we started FAQ section in here, I just added all in one page but let me know if you think it would make sense having separate. Otherwise happy to trim down the content, sorry about the redundancy

> ### What are Blob Parameter Only forks? {#what-are-blob-parameter-only-forks}

Reviewer: Okay yeah, makes sense ❤️ 

I guess my general take on "FAQs" is we should first attempt to provide quality thought-out content to users that would cover as much as possible (which I think we've done a good job of through the rest of this page so far). Then, if there are particularly common questions that people ask where a concise answer could be provided, an FAQ can be handy to fill in those gaps, or concisely and more directly answer this frequent questions. I'd be cautious though of effectively using the FAQ approach for larger amounts of content.

Curious your take on these thoughts

### What EVM changes are implemented? {#what-evm-changes-are-implemented}

Fusaka solidifies EVM with new minor changes and features. 

- For security while scaling, a maximum size of a single transaction will be [limited to 16.7 million](https://eips.ethereum.org/EIPS/eip-7825) units of gas. 
- [New opcode CLZ](https://eips.ethereum.org/EIPS/eip-7939), Count Leading Zeros, is added to EVM and will enable smart contract languages to perform certain operations more efficiently.
- [The cost ModExp precompile will be increased](https://eips.ethereum.org/EIPS/eip-7883), contracts using it will charge more gas for execution.

### How does new 16M gas limit affects contract developers? {#how-does-new-16m-gas-limit-affects-contract-developers}

> Fusaka introduces a limit to [maximum size of a single transaction to 16.7 million](https://eips.ethereum.org/EIPS/eip-7825) (2^24) gas units. This is roughly the size of an average block before which makes it big enough to accommodate even complex transactions that would consume an entire block. This limit creates protection for clients, preventing potential DoS attacks in the future with higher block gas limit. The goal of scaling is to enable more transactions get into the blockchain without a single one consuming all the block. 

Reviewer: ```suggestion
Fusaka introduces a limit to [maximum size of a single transaction to 16.7 million](https://eips.ethereum.org/EIPS/eip-7825) (2^24) gas units. This is roughly the previous size of an average block which makes it big enough to accommodate complex transactions that would consume an entire block. This limit creates protection for clients, preventing potential DoS attacks in the future with higher block gas limit. The goal of scaling is to enable more transactions to get into the blockchain without a single one consuming the whole block. 
```

> Regular user transactions are far from reaching this limit. Certain edge cases like big and complex Defi operations, large smart contract deployements or batch transactions targeting multiple contracts might be affected by this change. These transaction will have to be divided into smaller ones or optimized in another way. Use simulation before submitting transactions that potentially reach the limit.

Reviewer: ```suggestion
Regular user transactions are far from reaching this limit. Certain edge cases like big and complex DeFi operations, large smart contract deployments or batch transactions targeting multiple contracts might be affected by this change. These transaction will have to be divided into smaller ones or optimized in another way. Use simulation before submitting transactions that potentially reach the limit.
```

> RPC method `eth_call` is not limited and will allow simulation of bigger transactions than the actual blockchain limit. The actual limit for RPC methods can be configured by the client operator to ensure prevent abuse. 

Reviewer: ```suggestion
The RPC method `eth_call` is not limited and will allow simulation of bigger transactions than the actual blockchain limit. The actual limit for RPC methods can be configured by the client operator to ensure prevent abuse. 
```

### What CLZ means for developers? {#what-clz-means-for-developers}

> EVM compilers like Solidity will implement and utilize the new function for counting zeroes under the hood. New contracts might benefit from gas savings if they rely on this sort of operations. Follow releases and feature announcement of the smart contract language for documentation on potential savings. 

Reviewer: ```suggestion
EVM compilers like Solidity will implement and utilize the new function for counting zeros under the hood. New contracts might benefit from gas savings if they rely on this sort of operation. Follow releases and feature announcement of the smart contract language for documentation on potential savings. 
```

### Are there any changes for my existing smart contracts? {#what-clz-means-for-developers}

> Fusaka has no direct affect that would break any existing contracts or change their behavior. Changes introduced to execution layer are made with backwards compatibility, however, always keep an eye on edge cases and potential impact. 

Reviewer: ```suggestion
Fusaka has no direct affect that would break any existing contracts or change their behavior. Changes introduced to the execution layer are made with backward compatibility, however, always keep an eye on edge cases and potential impact. 
```

> [With the increased cost of ModExp precompil](https://eips.ethereum.org/EIPS/eip-7883), contracts that depend on it will charge more gas for execution. If your contract relies heavily on this and becomes more expensive for users, reconsider how it’s utilized. 

Reviewer: ```suggestion
[With the increased cost of `ModExp` precompile](https://eips.ethereum.org/EIPS/eip-7883), contracts that depend on it will consume more gas for execution. If your contract relies heavily on this and becomes more expensive for users, reconsider how it’s utilized. 
```
